src/main/java/esi/backend/service/RequestService.java

```diff
@@ -20,30 +23,61 @@ public class RequestService {
 
     @Autowired
     private RequestRepository requestRepository;
-
+    @Autowired
+    private RentalRepository rentalRepository;
+    @Autowired
+    private CarRepository carRepository;
     @Autowired
     private CustomerRepository customerRepository;
 
-    public List<Request> getAllRequests() {
-        List<Request> requests = new ArrayList<>();
-        requestRepository.findAll().forEach(requests::add);
-        return requests;
+    public ResponseEntity<List<Request>> getAllRequests() {
+        return new ResponseEntity<>(requestRepository.findAll(), HttpStatus.OK);
     }
 
-    public List<Request> getAllRequestsByCarId(UUID carId) {
-        return requestRepository.findByCarId(carId);
+    public ResponseEntity<List<Request>> getAllRequestsByCarId(UUID carId) {
+        Car car = carRepository.findById(carId).orElse(null);
+        if (car == null)
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        return new ResponseEntity<>(requestRepository.findByCarId(carId), HttpStatus.OK);
     }
 
-    public Optional<Request> getRequest(UUID id) {
-        return requestRepository.findById(id);
+    public ResponseEntity<Request> getRequest(UUID id) {
+        Request request = requestRepository.findById(id).orElse(null);
+        return (request == null) ? new ResponseEntity<>(HttpStatus.NOT_FOUND) : new ResponseEntity<>(request, HttpStatus.OK);
     }
 
-    public void addRequest(Request request) {
+    public void addRequest(UserDetails currentUser, Request request, UUID carId) {
+        Optional<Car> optionalCar = carRepository.findById(carId);
+        if (optionalCar.isPresent()) {
+            request.setCar(optionalCar.get());
+        } else throw new ResourceNotFoundException("Car with id" + carId + "not found");
+        Optional<Customer> optionalCustomer = customerRepository.findByUsername(currentUser.getUsername());
+        optionalCustomer.ifPresent(request::setCustomer);
         requestRepository.save(request);
     }
 
-    public void updateRequest(Request request) {
-        requestRepository.save(request);
+    public void updateRequest(UserDetails user, Request request, UUID id) {
+        Request req = requestRepository.findById(id)
+                .orElseThrow(() -> new ResourceNotFoundException("Request with id " + id + " not found"));
```

Comment:
In InvoiceService I did so:
```
Rental rental = rentalRepository.findById(rentalId).orElse(null);
  if (rental == null) {
    new ResponseEntity<>(HttpStatus.NOT_FOUND);
    return;
  }
```
in all void methods. I think it's better for consistency as a ResponseEntity is then returned in any case.